Update image-mailto.md (#2397)
enkidevs/curriculum: web/html/links-ii/image-mailto.md

# Mailto:

---

## Content

> Although links typically navigate to other pages, using the anchor `mailto:` attribute in the path will open the user's default email program on their computer. The link will also auto-load the assigned email address making it easy for user's to quickly send an email.

Reviewer: ```suggestion
Although links typically navigate to other pages, using the anchor `mailto:` attribute in the path will open the user's default email program on their computer. 

The link will also auto-load the assigned email address making it easy for user's to quickly send an email.
```

[View CodePen](https://codepen.io/enkidevs/pen/ejvqXx)

> `mailto:` has 4 attributes; cc, bcc, subject, and body.

Reviewer: I think these are called "header fields"
```suggestion
`mailto:` can have four *header fields*: `cc`, `bcc`, `subject`, and `body`.
```

> Attributes are added by writing their name followed by an equal sign and the attribute's value.

Reviewer: ```suggestion
Header fields are added by writing their name followed by an equal sign and the field's value.
```

> > 💡 When adding the subject and/or the body, rather than using spaces between words it is recommended to write `%20`[1] in place of each space. 

Reviewer: I think we can add a small phrase here which explains why it's not good to include spaces, and if users are more interested they can open the footnote

I've also suggested updating the footnote's headline to URL Syntactic Rules

```suggestion
> 💡 When adding the subject and/or the body, rather than using spaces between words it is recommended to write `%20` in place of each space. This is because URLs have strict syntactic rules[1].
```

> The order of the attributes does not matter. However, when using more than one attribute is important to separate each of them with a `&` (and) sign. The main e-mail is always separated with a `?` from the other field values.

Reviewer: ```suggestion
The order of the attributes does not matter. However, when using more than one attribute it is important to separate each of them with a `&` (and) sign.

The main e-mail is always separated with a `?` from the other field values.
```

Create a text link that opens the user's default email program.

> ```html
> <??? ???="??? ???">Email Me<???>
> ```

Reviewer: formatting it like this will ensure questions gaps don't get pushed off-screen 

also, `mailto:` and `heya@enkipro.com` shouldn't have a space between them. I've suggested it like this `??????` but we should test this quickly in spider before merging (i.e. make a new workout/insight with the pq having two question gaps `??????` together)

```suggestion
```html
<??? ???="??????">
  Email Me
<???>
```
```

Author: If I remember correctly, two question gaps put together work fine. In the app, they are shown with a spacing smaller than a space between them. I know there are a few insights that already have them, I will check them right now to see if they work fine

Author: They work, there is no spacing but there is a clear visible cue that shows they are separate boxes. I am sending it to you on slack

- mail:
- address:
- email:

---
## Footnotes

> [1:%20 instead of space in a URL]

Reviewer: ```suggestion
[1:URL Syntactic Rules]
```

> The reason for replacing spaces with `%20` is because URLs have strict syntactic rules. URLs can't have spaces, so the ASCII character for space should be inserted instead. The `%` character, followed by two hexadecimal digits, is used to replace unsafe characters from a URL. The ASCII character for space is `20`. Making `%20` the replacement character for space.

Reviewer: we can split this paragraph into multiple smaller ones to make it a bit more readable

```suggestion
The reason for replacing spaces with `%20` is because URLs have strict syntactic rules. 

URLs can't have spaces, so the ASCII character for space should be inserted instead. 

The `%` character, followed by two hexadecimal digits, is used to replace unsafe characters from a URL. 

The ASCII character for space is `20`, making `%20` the replacement character for space.
```
